AnnoDesigner/MainWindow.xaml

```diff
@@ -126,8 +124,15 @@
                 <MenuItem Header="{l:Localize Language}"
                           TabIndex="23"
                           ItemsSource="{Binding Languages}">
+                    <MenuItem.Resources>
```

Comment:
Could you add a border around the image, so Polish flag isn't visible only as a red stripe?

Reply:
1. The menu is gray and the flag is white/red.
2. I tried, but it doesn't work. The image is missing sometimes.
My code:
```xaml
<MenuItem Header="{l:Localize Language}"
          TabIndex="23"
          ItemsSource="{Binding Languages}">
    <MenuItem.Resources>
        <Image x:Key="flagImage"
               Source="{Binding FlagPath}"
               x:Shared="False" />
    </MenuItem.Resources>
    <MenuItem.ItemContainerStyle>
        <Style TargetType="MenuItem">
            <Setter Property="Icon">
                <Setter.Value>
                    <Border BorderThickness="1"
                            BorderBrush="Black">
                        <ContentPresenter Content="{StaticResource flagImage}" />
                    </Border>
                </Setter.Value>
            </Setter>
            <Setter Property="Header"
                    Value="{Binding Name}" />
            <Setter Property="IsCheckable"
                    Value="True" />
            <Setter Property="IsChecked"
                    Value="{Binding IsSelected, Mode=TwoWay}" />
            <Setter Property="Command"
                    Value="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}, Path=DataContext.LanguageSelectedCommand}" />
            <Setter Property="CommandParameter"
                    Value="{Binding}" />
        </Style>
    </MenuItem.ItemContainerStyle>
</MenuItem>
```

Reply:
I've found a solution for this. The key idea is to make sure that the full content that you need is declared in `Resources` with `x:Shared="false"`. For example,

```xml
<MenuItem Header="{l:Localize Language}"
          TabIndex="23"
          ItemsSource="{Binding Languages}">
    <MenuItem.Resources>
        <Image x:Key="flagImage"
               Source="{Binding FlagPath}"
               x:Shared="False" />
        <Border BorderThickness="1"
                BorderBrush="Black" x:Key="iconWithBorder" x:Shared="false">
            <ContentPresenter Content="{StaticResource flagImage}" />
        </Border>
    </MenuItem.Resources>
    <MenuItem.ItemContainerStyle>
        <Style TargetType="MenuItem">
            <Setter Property="Icon" Value="{StaticResource iconWithBorder}" />
            <Setter Property="Header"
                    Value="{Binding Name}" />
            <Setter Property="IsCheckable"
                    Value="True" />
            <Setter Property="IsChecked"
                    Value="{Binding IsSelected, Mode=TwoWay}" />
            <Setter Property="Command"
                    Value="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}, Path=DataContext.LanguageSelectedCommand}" />
            <Setter Property="CommandParameter"
                    Value="{Binding}" />
        </Style>
    </MenuItem.ItemContainerStyle>
</MenuItem>

```